### PR TITLE
Add Gmail specific info in setup.

### PR DESCRIPTION
**Pull request by t-oster**
This PR adds a GMail specific text with links to `enable less secure apps` or `create an app password` to the setup dialog, if the E-Mail adress is recognized as GMail (`@gmail.` or `@googlemail.`).
It addresses #20 (#178).

screenshot_20180622-073030

### CODE REVIEW

**Comment by r10s** [comment]
great that you push this forward :)

i've just added a commit that allows the advanced options to be expanded (the onClickListener was no longer called due to LinkMovementMethod).

however, i would suggest to simplify this dialog a bit:
- merge MyAccountExplain2 into MyAccountExplain and show it directly after the password field (so that there is only one place for a hint)
- MyAccountExplain could be replaced _completely_ by MyAccountExplainGmail using the new logic
- instead of the "+" icon i would add a button  "Show more options..." (or a link or so) below the explaining text (i've seen more than one person not getting that the "+" is a button)

what do you think?

**Comment by t-oster** [comment]
agreed

**Comment by r10s** [comment]
great, if you run into troubles with the weird settings objects, please let me know.

as the "+" icon and the gmail-text would share the same item, we should replace it with a "more options ..:" link, so we won't get in trouble with link-click vs. onclick-handler.

**Comment by t-oster** [comment]
yes

**Comment by violoncelloch** [comment]
Wouldn't it be more clear to place the + plus on a white space to indicate it's a collapsed input area?

**Comment by t-oster** [comment]
screenshot_20180623-083636
screenshot_20180622-073030
screenshot_20180623-083621
screenshot_20180623-083631

**Comment by t-oster** [comment]
To-Do:
- [ ] make `advanced opations` (at least Look Like) a Button
- [ ] remove whitespace-changes
- [ ] fix german Translation
- [ ] squash everything to one commit

**Comment by r10s** [comment]
great :)

> make advanced opations (at least Look Like) a Button

i would just made the field look like a link (same as the links on the second screenshot) and remove the "+".

**Comment by r10s** [comment]
> I think it would be prettier and easier to understand if we place "advanced settings... +" on white background (input area) to make it clear it's an expandable input location

i would be fine with both. my general idea is that these "advanced settings" should not be needed for the majority of users, so it seems okay if it's not too disturbing. but a normal black-on-white button below the explaining text seems to be also very okay :) and would have the advantage that this is a know element.

**Comment by violoncelloch** [comment]
Ok, if we see it this way a link or something less obvious seems ok too

**Comment by t-oster** [comment]
While i'm at it, is there another provider needing special Info?

**Comment by testbird** [comment]
Possible source: https://github.com/deltachat/deltachat-core/wiki/Provider-overview

**Comment by testbird** [comment]
Don't remember the provider, but someone could not connect using DC right away.
Only later he found the provider had send him an email to confirm enabling email client access.
So maybe you could add to the general hint to check the emails in the usual way if login always fails despite the correct credentials.

**Comment by violoncelloch** [comment]
@t-oster I think a problem could be that corporates using business domains but google or other services in the background are not detected as google mail through the actual algorythem. Are imap settings automatically detected through /.well-known or DNS Records for googles imap and smtp servers? So maybe it would be useful to provide these extra infos after the detection of a google imap server as well even though it's not `@gmail.` or `@googlemail.`

**Comment by testbird** [comment]
Generally inform about leaking your contacts including their IPs to `$PROVIDER` ?

**Comment by t-oster** [comment]
@testbird what exactly are you referring to?

**Comment by t-oster** [comment]
@violoncelloch I don't think we currently get back the autoconfig values, so I am not sure if I can detect business domains with gmail backend. I could however also change the text if the user manually inputs a gmail smtp/imap server. However at this stage, the user is already below the `advanced settings` area and probably knows what he's doing(?).
Do you have an idea on how many people are using gmail with business accounts? Do you have an example in order to check if autoconfig even works?

**Comment by t-oster** [comment]
The PR is now rebased and squashed.

20180625080843_image
20180625080844_image
20180625080842_image

**Comment by violoncelloch** [comment]
@t-oster 
I agree, if the user is already in the advanced settings he most probably knows what he's doing and how to handle thes gmail things, so this doesn't make much sense...
 I don't have an account to test with. I only know there are quite a few corporates relying on google services.

**Comment by r10s** [comment]
thanks, looks great :)

i've slightly modified the button in https://github.com/deltachat/deltachat-android/commit/8db7c9f45b512972138d125108962a4007188ff5 , must admit that a normal-looking button may be the better choice :)

screenshot_20180626-015824

**Comment by testbird** [comment]
> what exactly are you referring to?

Was thinking about a GDPR complient notice. Something like
```
By using (not self-hosted) email providers, you transmit personal data 
(including IP addresses) to the email provider, and the GDPR requires 
you to disclose this to your contacts.

Your current email provider is: X operated by Y.
```

**Comment by t-oster** [comment]
well. I think this is a bit off topic, because this issue was to help users getting DC to work and not to tell the user what he should know when using E-mail in general. Also I am not sure if the GDPR applies to private E-Mail users. At least I did not see such notice in any other MUA.

**Comment by testbird** [comment]
Hi, it would sure be another issue. The comments / violoncelloch just touched on that more general issue.

When programing your own MUA it may not be that relevant. But if software is to be used by others, especially in business, users need to know what data is transfered.

It would be a good thing if DC would also be first to meet this. New issue?

**Comment by t-oster** [comment]
You can open a new issue if you think this is relevant. I would not add this warning in the configuring section, because it would confuse users. Maybe we have to add a generic "Datenschutzerklärung" but since we are not the provider, I think this is not necessary and I don't think any other E-Mail client will add such a notice. It is obvious to me that if I send an E-Mail, it contains the E-Mail Address of the receipient.... I also don't add a notice to my snail-mail regular letters to inform the user that the postmaster will read it.
Also IMHO I do not send the IP address of the receipient to google, but the receipient does when answering the mail, so his MUA should warn him.

So long story short: IMHO we have much more important stuff to do than adding confusing and obvious information, but I may be wrong.

**Comment by testbird** [comment]
I agree that coding wise there may be more important stuff, nevertheless, users have legal obligations, not being aware can easily happen, but is no excuse. DC could just be helpful and make a good example.

>  It is obvious to me that if I send an E-Mail, it contains the E-Mail Address of the receipient....

And yours, paired with your IP! Allowing any obvious intermediate providers (and those hidden under a different flag) to personalize your -analytics.

A comment above noted that the email domain does not always correspond with the actual provider.

**Comment by testbird** [comment]
Thanks for the link, very good. I propose an attempt to sort the bullets into more easily comprehensible points https://github.com/deltachat/deltachat-pages/pull/89

It boils down to exactly the same crucial point: the email provider.

However, as users have no control about the email provider that others (incl. individual private users) use to contact them, and any reply would already transmit data, the only privacy-improving way I see would be to let the email client inform the user about the providers involved on the clients side, especially if they differ from the email domain.

**Comment by testbird** [comment]
@hpk42 The messages above were the reason for me reading and creating https://github.com/deltachat/deltachat-pages/pull/89
